superset/views/core.py

```diff
@@ -2528,6 +2530,43 @@ def sql_json(self):
             json_error_response(
                 'Database with id {} is missing.'.format(database_id))
 
+        # Validation request.
+        if validate_only:
```

Comment:
Why not a different endpoint for validation?

Reply:
No particular reason, apart from an instinct to try to minimize the changeset — I'm happy to pull it out if you'd prefer it that way!

Reply:
Pulled out to a separate endpoint this morning.